cloud_function/migrate_database.py

```
@@ -0,0 +1,586 @@
+                if stats["inserted"] > 0:
+                    changes.append(f"{stats['inserted']} inserted")
+                if stats["updated"] > 0:
+                    changes.append(f"{stats['updated']} updated")
+                if stats["deleted"] > 0:
+                    changes.append(f"{stats['deleted']} deleted")
+
+                log(f"  [OK] Changes: {', '.join(changes)}")
+            elif stats.get("error"):
+                # Track error
+                sync_errors.append(f"{table_name}: {stats['error'][:100]}")
+                log(f"  [WARNING] Error detected")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)